### PR TITLE
[ListTile] adds new properties to customize the tile color

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Adds two new properties, `tileColor` and `selectedTileColor` respectively to change the tile color of `ListTile`. 

Usage

- `ListTile.selected` is true.
```dart
ListTile(
  selected: true,
  selectedTileColor: Colors.brown,
  tileColor: Colors.black,
  title: const Text(
    'Title',
     style: TextStyle(color: Colors.orange),
   ),
)
```

tile_color_when_selected_is_true

- `ListTile.selected` is false.

```dart
ListTile(
  selected: false,
  selectedTileColor: Colors.brown,
  tileColor: Colors.black,
  title: const Text(
    'Title',
     style: TextStyle(color: Colors.orange),
   ),
)
```
tile_color_when_selected_is_false

## Related Issues
#33392

## Tests

I added the following tests:

- Added test to check that `ListTile` respects `tileColor` and `selectedTileColor`.
- Added test to check the default tile color of `ListTile`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
  

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/list_tile.dart [hunk 1]
```diff
@@ -808,6 +810,16 @@ class ListTile extends StatelessWidget {
   /// {@macro flutter.widgets.Focus.autofocus}
   final bool autofocus;
 
+  /// Defines the background color of `ListTile.
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/list_tile.dart:813]:
```suggestion
  /// Defines the background color of `ListTile` when [selected] is false.
```

File: packages/flutter/test/material/list_tile_test.dart [hunk 2]
```diff
@@ -1534,4 +1534,80 @@ void main() {
...
+                  onTap: () {
+                    setState(()=> isSelected = !isSelected);
+                  },
+                  title: const Text('Title'),
+                );
+              },
+            ),
+          ),
+        ),
+      )
```
Comment (shihaohong) [inline packages/flutter/test/material/list_tile_test.dart:1562]: nit:
```suggestion
      ),
```

File: packages/flutter/test/material/list_tile_test.dart [hunk 3]
```diff
@@ -1534,4 +1534,80 @@ void main() {
...
+                    onTap: () {
+                      setState(()=> isSelected = !isSelected);
+                    },
+                    title: const Text('Title'),
+                  );
+                },
+              ),
+            ),
+          ),
+        )
```
Comment (shihaohong) [inline packages/flutter/test/material/list_tile_test.dart:1599]: nit: indentations and trailing commas
```suggestion
      MaterialApp(
        home: Material(
          child: Center(
            child: StatefulBuilder(
              builder: (BuildContext context, StateSetter setState) {
                return ListTile(
                  selected: isSelected,
                  onTap: () {
                    setState(()=> isSelected = !isSelected);
                  },
                  title: const Text('Title'),
                );
              },
            ),
          ),
        ),
      ),
```

**Review by shihaohong — COMMENTED** [review]
Some code nits, otherwise the implementation and the tests look good

**Review by shihaohong — APPROVED** [review]
LGTM

**Comment by kakyoism** [comment]
Is this available in other channel like dev?

**Pull request merged**